src/API.jl

```diff
@@ -224,7 +233,13 @@ function develop(ctx::Context, pkgs::Vector{PackageSpec}; shared::Bool=true,
 end
 
 function add(ctx::Context, pkgs::Vector{PackageSpec}; preserve::PreserveLevel=PRESERVE_TIERED,
-             platform::AbstractPlatform=HostPlatform(), kwargs...)
+             platform::AbstractPlatform=HostPlatform(), installed::Union{Bool,Nothing}=nothing, kwargs...)
+
+    if installed === true && !in(preserve, (PRESERVE_TIERED, PRESERVE_ALL_INSTALLED))
```

Comment:
Could this be moved to `tiered_resolve` so it doesn't have to be repeated in two places (both `develop` and `add`).?